### PR TITLE
iox-#1196 Add lint suppressions and fix warnings for PoorMansHeap

### PR DESCRIPTION
**Pull request by elBoberido**
## Pre-Review Checklist for the PR Author

1. [x] Code follows the coding style of [CONTRIBUTING.md][contributing]
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-#123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit messages are signed (`git commit -s`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
    - [x] Each unit test case has a unique UUID
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates to #1196

### CODE REVIEW

File: iceoryx_hoofs/include/iceoryx_hoofs/cxx/poor_mans_heap.hpp [hunk 1]
```diff
@@ -31,6 +31,9 @@ class PoorMansHeapType
 {
 };
 
+constexpr uint64_t POOR_MANS_HEAP_DEFAULT_ALIGNMENT{8};
+
+// AXIVION Construct Ruleset-A12.1.1 : it is guaranteed that the memory is initialized on access
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/cxx/poor_mans_heap.hpp:36]: You mean the AUTOSAR rule, correct?

```suggestion
// AXIVION Construct AutosarC++19_03-A12.1.1 : it is guaranteed that the memory is initialized on access
```

Comment (elBoberido) [reply iceoryx_hoofs/include/iceoryx_hoofs/cxx/poor_mans_heap.hpp:36]: :grimacing: correct

**Review by FerdinandSpitzschnueffler — APPROVED** [review]

File: iceoryx_hoofs/include/iceoryx_hoofs/cxx/poor_mans_heap.hpp [hunk 2]
```diff
@@ -31,6 +31,9 @@ class PoorMansHeapType
 {
 };
 
+constexpr uint64_t POOR_MANS_HEAP_DEFAULT_ALIGNMENT{8};
+
+// AXIVION Construct AutosarC++19_03-A12.1.1 : it is guaranteed that the memory is initialized on access
```
[the remark this answers is not in the document]
Comment (mossmaurice) [reply iceoryx_hoofs/include/iceoryx_hoofs/cxx/poor_mans_heap.hpp:36]: Oh, I just realized that `Construct` is only activae till `};` so you need to move it to the `.inl`

**Review by elfenpiff — DISMISSED** [review]

File: iceoryx_hoofs/include/iceoryx_hoofs/cxx/stack.hpp [hunk 3]
```diff
@@ -23,7 +23,7 @@ namespace iox
 {
 namespace cxx
 {
-// AXIVION Construct Ruleset-A12.1.1 : it is guaranteed that the array elements are initialized on access
+// AXIVION Construct AutosarC++19_03-A12.1.1 : it is guaranteed that the array elements are initialized on access
```
Comment (mossmaurice) [inline iceoryx_hoofs/include/iceoryx_hoofs/cxx/stack.hpp:26]: @elBoberido I don't get `A12.1.1` reported in `stack` when removing this rule and running the check locally.

@dkroenke neillangmead41 Could you check that?

Comment (elBoberido) [reply iceoryx_hoofs/include/iceoryx_hoofs/cxx/stack.hpp:26]: Just to be on the same page which warning is shown on the dashboard
image

cc @mossmaurice @saif-at-github @dkroenke

**Pull request merged**